prescient/engine/egret/egret_plugin.py

```diff
@@ -1309,33 +1045,65 @@ def solve_deterministic_day_ahead_pricing_problem(solver, ruc_results, options,
+
+    Parameters
+    ----------
+    options:Options
+        Global option values
+    data_provider: DataProvider
+        An object that can provide actual and/or forecast data for the requested days
+    this_date: string
+        A string that can be parsed as a date
+    this_hour: int
+        0-based index of the first hour of the day for which data should be retrieved
+    ''' 
+    # Convert time string to time
+    start_time = dateutil.parser.parse(this_date)
+
+    # Pick whether we're getting actuals or forecasts
+    if options.simulate_out_of_sample:
+        get_data_func = data_provider.populate_with_actuals
+    else:
+        if options.run_deterministic_ruc:
+            print("")
+            print("***WARNING: Simulating the forecast scenario when running deterministic RUC - "
+                  "time consistency across midnight boundaries is not guaranteed, and may lead to threshold events.")
+        get_data_func = data_provider.populate_with_forecast_data
 
-        if not os.path.exists(simulated_dat_filename_next):
-            raise RuntimeError("The file " + simulated_dat_filename_next + " does not exist or cannot be read.")
+    # Get a new model
+    # TODO: Avoid hard-coding the number of values
+    md = data_provider.get_initial_model(options, 48)
 
-        ruc_instance_to_simulate_next_period = uc_abstract_data_model.create_instance(simulated_dat_filename_next)
+    # Fill it in with data
+    data_provider.populate_initial_state_data(options, start_time.date(), md)
+    if this_hour == 0:
+        get_data_func(options, start_time, 48, 60, md)
     else:
-        ruc_instance_to_simulate_next_period = None
+        get_data_func(options, start_time, 24, 60, md)
+        _copy_time_period_data(options, range(0,24), range(24,48), md)
 
-    print("")
-    print("Creating RUC instance to simulate")
-    new_ruc_to_simulate_md = _get_ruc_data(options, ruc_instance_to_simulate_this_period, this_hour, ruc_instance_to_simulate_next_period )
+    return md
 
-    return new_ruc_to_simulate_md
+def _copy_time_period_data(options, copy_from_periods, copy_to_periods, model):
+    ''' Copy forecastable data between time periods
+    '''
+    # copy renewables limits
+    for gen, gdata in model.elements('generator', generator_type='renewable'):
+        for t_from, t_to in zip(copy_from_periods, copy_to_periods):
+            gdata['p_min']['values'][t_to] = gdata['p_min']['values'][t_from]
+            gdata['p_max']['values'][t_to] = gdata['p_max']['values'][t_from]
+
+    # Copy loads
+    for bus, bdata in model.elements('load'):
+        for t_from, t_to in zip(copy_from_periods, copy_to_periods):
+            bdata['p_load']['values'][t_to] = bdata['p_load']['values'][t_from]
+
+    # Copy reserve requirement
+    reserve_reqs = model.data['system']['reserve_requirement']['values']
+    for t_from, t_to in zip(copy_from_periods, copy_to_periods):
+        reserve_reqs[t_to] = reserve_reqs[t_from]
```

Comment:
Similar to the comment above -- since these are carrying forward the same information I think it would be good to have it annotated in a single place.

Reply:
See issue #48